apps/api/src/main.ts

```diff
@@ -1,10 +1,44 @@
+import { ValidationPipe } from "@nestjs/common";
 import { NestFactory } from "@nestjs/core";
+import RedisStore from "connect-redis";
+import * as session from "express-session";
+import * as passport from "passport";
+import { createClient } from "redis";
 import { AppModule } from "./app.module";
 
 async function bootstrap() {
     const app = await NestFactory.create(AppModule);
     app.setGlobalPrefix("api");
+    app.useGlobalPipes(
+        new ValidationPipe({
+            transform: true,
+            whitelist: true,
+        }),
+    );
+
+    const redisClient = await createClient({
+        url: String(process.env.REDIS_URL),
```

Comment:
after searching found that nest themselves do the same as what I did here

Reply:
Yes there's no problem with this way, however, a config module is a cleaner way to manage configs.

Reply:
I know that but since this is the preferred one for nest then no problem doing so
I can implement a function that provides me the singleton object but this will not give us any advantage, so nno headache is better